gen/SimDevice.yml

```diff
@@ -39,65 +42,214 @@ functions:
       Get:
       Set:
   SimEnum:
-    shared_ptr: true
+    shared_ptr: false
     force_no_trampoline: true
     methods:
       SimEnum:
         overloads:
           "":
+            ignore: true
           HAL_SimValueHandle:
+            param_override:
+              val:
+                name: handle
       Get:
       Set:
   SimBoolean:
-    shared_ptr: true
+    shared_ptr: false
     force_no_trampoline: true
     methods:
       SimBoolean:
         overloads:
           "":
+            ignore: true
           HAL_SimValueHandle:
+            param_override:
+              val:
+                name: handle
       Get:
       Set:
   SimDevice:
-    shared_ptr: true
+    shared_ptr: false
+    force_type_casters:
+    - wpi::SmallVector
     methods:
       SimDevice:
         overloads:
           "":
+            ignore: true
           char*:
           char*, int:
           char*, int, int:
           HAL_SimDeviceHandle:
       CreateValue:
+        ignore: true
       CreateDouble:
       CreateEnum:
         overloads:
           char*, bool, std::initializer_list<const char *>, int32_t:
             ignore: true
           char*, bool, wpi::ArrayRef<const char *>, int32_t:
+            cpp_code: |
+              [](SimDevice * self, const char * name, bool readonly, const wpi::SmallVector<std::string, 8> &options, int32_t initialValue) {
+                wpi::SmallVector<const char *, 8> coptions;
+                coptions.reserve(options.size());
+                for (auto s: options) {
+                  coptions.push_back(s.c_str());
+                }
+                return self->CreateEnum(name, readonly, coptions, initialValue);
+              }
+
       CreateBoolean:
 
 inline_code: |
+
+  cls_SimValue
+    .def_property_readonly("value", [](const hal::SimValue &self) -> py::object {
+        HAL_Value value;
+        {
+          py::gil_scoped_release release;
+          value = self.GetValue();
+        }
+        switch (value.type) {
+        case HAL_BOOLEAN:
+          return py::bool_(value.data.v_boolean);
+        case HAL_DOUBLE:
+          return py::float_(value.data.v_double);
+        case HAL_ENUM:
+          return py::int_(value.data.v_enum);
+        case HAL_INT:
+          return py::int_(value.data.v_int);
+        case HAL_LONG:
+          return py::int_(value.data.v_long);
+        default: 
+          return py::none();
+        }
+    })
+    .def("__bool__", [](const hal::SimValue &self) -> bool {
+      return (bool)self;
+    })
+    .def("__repr__", [](const hal::SimValue &self) -> py::str {
+        if (!self) {
+          return "SimValue(<invalid>)";
+        }
+        HAL_Value value;
+        {
+          py::gil_scoped_release release;
+          value = self.GetValue();
+        }
+        switch (value.type) {
+        case HAL_BOOLEAN:
+          return std::string("SimValue(type=bool, value=") + (value.data.v_boolean ? "True" : "False") + ")";
```

Comment:
It's not valid to construct a SimValue like this right? IMO the repr should only look like Python code if that code would be a valid way to construct such an instance (otherwise an arbitrary string in `<>` would be fine).

Reply:
Well then you better remove all those units I added to trajectories... 

Actually, I wondered why your reprs all looked like that, since I usually prefer `<Foo x=1>`

Reply:
tl;dr I like what dataclasses do